app/models/access_control_list.rb

```diff
@@ -3,6 +3,7 @@ class AccessControlList < ActiveRecord::Base
   belongs_to :resource, polymorphic: true
 
   validates_presence_of :user_group
+  validates_uniqueness_of :user_group, scope: [:resource_id, :resource_type], message: ". Roles have already been set for this user or group"
```

Comment:
Yes it makes the error message look like "Validation Failed: User groups. Roles.."